pkg/objectstate/merge/merge.go

```diff
@@ -55,44 +55,5 @@ func MetadataForUpdate(current, updated client.Object) (client.Object, error) {
 	updated.SetManagedFields(current.GetManagedFields())
 	updated.SetFinalizers(current.GetFinalizers())
 
-	_ = Annotations(current, updated)
```

Comment:
This involve an actual change in the behavior - it won't keep the current labels, only the updated one.
What's the rational behind this change?
According to the commit message:
```
if the key is found in the current map it will
still exist in the updated one, while the required is to omit it.
```
But that means the user can override the labels, because the operator is no longer maintaining the existing ones.

Reply:
> it won't keep the current labels, only the updated one.

That's exactly the issue. The problem in the current flow of updating the annotations is that in the case where the user intentionally didn't add the annotation (equivalent to deleting it if it exists) the code won't update that in the next results the annotation will be removed. It'll still exist. Then how the user is supposed to delete that specific annotation?
The annotations are built at every reconciliation iteration, and the RTE ObjectState carry this data until it gets ignored in this MetadataForUpdate.

Reply:
As far as I have checked, updating the annotations never happened so far that is why this problem wasn't visible before this PR

Reply:
I see your point, but this how we check for the annotation:
https://github.com/openshift-kni/numaresources-operator/blob/27a91fd43bd576f719a935fe2a5d54ed18a5e071/internal/api/annotations/annotations.go#L29 
so in case we want to disable it, we can change the value to be different than `custom`

Second thing, assuming we're not keeping the existing one like this PR suggests, and one day we'll add another annotation (to enable some other functionality).
If we would like to change one of the annotations we'll have to send all the other annotations as well, so the operator won't delete the existing ones.

@ffromani Could you please provide your POV here?